core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -160,18 +182,6 @@ public static Suggest readSuggest(StreamInput in) throws IOException {
         return result;
     }
 
-    public static Map<String, List<Suggest.Suggestion>> group(Map<String, List<Suggest.Suggestion>> groupedSuggestions, Suggest suggest) {
-        for (Suggestion<? extends Entry<? extends Option>> suggestion : suggest) {
-            List<Suggestion> list = groupedSuggestions.get(suggestion.getName());
-            if (list == null) {
-                list = new ArrayList<>();
-                groupedSuggestions.put(suggestion.getName(), list);
-            }
-            list.add(suggestion);
-        }
-        return groupedSuggestions;
-    }
-
     public static List<Suggestion<? extends Entry<? extends Option>>> reduce(Map<String, List<Suggest.Suggestion>> groupedSuggestions) {
```

Comment:
we should extract the reduce logic from suggest, the suggest abstraction is not the right place for these, maybe somewhere in `SearchPhraseController`? I think this will simplify `Suggest` and help with nuking the crazy generics. maybe we should simplify this in a subsequent PR?